### PR TITLE
Added non-inline HyperLink generation for 'pdf' module.

### PR DESCRIPTION
**Pull request by BilakshanP**
+ Added main.rs to .gitignore for local testing purposes.
+ Added non-inline HyperLink generation to 'pdf' module.
    + Non-inline (i.e In-text) hyperlinks are not supported yet.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ Cargo.lock
 **/*.rs.bk
 *.pdb
 lib/test/data/generated*
+lib/src/main.rs
```
Comment (evgenyigumnov) [inline .gitignore:8]: Probably it should be removed

**Comment by evgenyigumnov** [comment]
Look at https://github.com/igumnoff/shiva/issues/4 please

Do you see "2. Units test requered for this new feature"

I do not see any tests about hyperlink feature

File: lib/src/pdf.rs [hunk 2]
```diff
@@ -416,14 +417,68 @@ impl TransformerTrait for Transformer {
...
+                        Some(printpdf::HighlightingMode::Invert),
+                    ));
+                }
+
                 _ => {}
             }
 
             Ok(())
         }
+
```
Comment (evgenyigumnov) [inline lib/src/pdf.rs:478]: Look at https://github.com/igumnoff/shiva/issues/4 please

Do you see "2. Units test requered for this new feature"

I do not see any tests about hyperlink feature

**Review by evgenyigumnov — CHANGES_REQUESTED** [review]

**Comment by evgenyigumnov** [comment]
1. test stdout

```
==========================
Document { elements: [Paragraph { elements: [Text { text: "Line 1", size: 8 }, Text { text: "Line 2", size: 8 }, Text { text: "Line 3", size: 8 }] }, Hyperlink { title: "Example", url: "https://www.example.com", alt: "Example Site" }, Hyperlink { title: "GitHub", url: "https://www.github.com", alt: "GitHub" }], page_width: 210.0, page_height: 297.0, left_page_indent: 10.0, right_page_indent: 10.0, top_page_indent: 20.0, bottom_page_indent: 10.0, page_header: [], page_footer: [] }
==========================
8
8
8

```
could you remove prinln with 8 ? :)

2. could you add property "size" in hyperlink?

**Comment by BilakshanP** [comment]
>     2. could you add property "size" in hyperlink?

I think size can be an Option<u8>, like while parsing markdown documents it can be None (default) for simple text & depending upon the headings used it could be any other value.

And while parsing pdf it'll get a definite value from the document. How does that sounds?

OR

We can have a separate property-struct called "formatting" which will have 'size', 'bold', 'underline', 'italics' etc property in it. (maybe a future suggestion)

**Comment by evgenyigumnov** [comment]
```
igumn@lenovo MINGW64 ~/shiva (issue4)
$ cargo test
   Compiling utf8parse v0.2.1
   Compiling anstyle v1.0.6
   Compiling anstyle-query v1.0.2
   Compiling colorchoice v1.0.0
   Compiling strsim v0.11.1
   Compiling clap_lex v0.7.0
   Compiling shiva v0.2.3 (C:\Users\igumn\shiva\lib)
   Compiling anstyle-parse v0.2.3
   Compiling anstyle-wincon v3.0.2
   Compiling anstream v0.6.13
warning: unused variable: `images`
 --> lib\src\json.rs:7:32
  |
7 |     fn parse(document: &Bytes, images: &HashMap<String, Bytes>) -> anyhow::Result<Document> {
  |                                ^^^^^^ help: if this is intentional, prefix it with an underscore: `_images`
  |
  = note: `#[warn(unused_variables)]` on by default

   Compiling clap_builder v4.5.2
error[E0063]: missing field `size` in initializer of `core::Element`
   --> lib\src\pdf.rs:880:17
    |
880 |                 Hyperlink {
    |                 ^^^^^^^^^ missing `size`

error[E0063]: missing field `size` in initializer of `core::Element`
   --> lib\src\pdf.rs:885:17
    |
885 |                 Hyperlink {
    |                 ^^^^^^^^^ missing `size`

For more information about this error, try `rustc --explain E0063`.
error: could not compile `shiva` (lib test) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
warning: `shiva` (lib) generated 1 warning (run `cargo fix --lib -p shiva` to apply 1 suggestion)
```

**Pull request merged**